blockchain/validation.go

```diff
@@ -83,13 +178,29 @@ func (bc *BlockChain) ValidBlock(b *Block) (bool, BlockCode) {
 		return false, BadBlockNumber
 	}
 
+	// Check that the first transaction is a CloudBase transaction
+	if valid, _ := ValidCloudBase(b.Transactions[0]); !valid {
+		return false, BadCloudBaseTransaction
+	}
+
 	// Verify every Transaction in the block.
-	for _, t := range b.Transactions {
+	for _, t := range b.Transactions[1:] {
 		if valid, _ := bc.ValidTransaction(t); !valid {
 			return false, BadTransaction
 		}
 	}
 
+	// Check that the target is within the min and max difficulty levels
+	target := HashToBigInt(b.Target)
+	if target.Cmp(MinDifficulty) == 1 || target.Cmp(MaxDifficulty) == -1 {
+		return false, BadTarget
+	}
+
+	// Check that time is not greater than current time or equal to 0
+	if uint32(b.Time) == 0 || uint32(b.Time) > uint32(time.Now().Unix()) {
```

Comment:
Not really. I think it's something to consider later and for now we can just not worry about checking time.